src/functions/module_lookup.ts

```diff
@@ -858,3 +860,122 @@ export const PIVOT = {
   },
   returns: ["RANGE<ANY>"],
 } satisfies AddFunctionDescription;
+
+//--------------------------------------------------------------------------
+// OFFSET
+//--------------------------------------------------------------------------
+
+export const OFFSET = {
+  description: _t(
+    "Returns a range reference shifted a specified number of rows and columns from a starting cell reference."
+  ),
+  args: [
+    arg(
+      "cell_reference (meta)",
+      _t("The starting point from which to count the offset rows and columns.")
+    ),
+    arg("offset_rows (number)", _t("The number of rows to offset by.")),
+    arg("offset_columns (number)", _t("The number of columns to offset by.")),
+    arg(
+      "height (number, default='height of cell_reference')",
+      _t("The height of the range to return starting at the offset target.")
+    ),
+    arg(
+      "width (number, default='width of cell_reference')",
+      _t("The width of the range to return starting at the offset target.")
+    ),
+  ],
+  compute: function (
+    cellReference: Maybe<{ value: string }>,
+    offsetRows: Maybe<FPayload>,
+    offsetColumns: Maybe<FPayload>,
+    height: Maybe<FPayload>,
+    width: Maybe<FPayload>
+  ): Matrix<FPayload> {
+    if (isEvaluationError(cellReference?.value)) {
+      throw cellReference;
+    }
+
+    const _cellReference = cellReference?.value;
+    const zone = toZone(_cellReference!);
+
+    let offsetHeight = zone.bottom - zone.top;
```

Comment:
I think the variable is not properly named. The height/width of the value returned will never be 0, so instead, I would set `let offsetHeight = zone.bottom - zone.top +1` such that it describes exactly what it does and remove the +1 in specific places, like when we compute `dependencyZone`

Reply:
why can't the height/width be 0 ?

Reply:
Based on the variable name, I would expect this value to be one of the dimension of the returned matrix of results. for instance in a matrix with `m` rows, I would expect offsetHeight to be `m`. I think we can agree that having a matrix with one of its dimension being 0 doesn't make sense^^.
But in the code you wrote, offsetHeight is set to `m-1` and then you call `generateMatrix` by incrementing the value of offsetHeight.

My comment was more about the code readability, to me the variable name is misleading, hence my suggestion.